File: app/src/main/java/com/android/camera/activity/MainActivity.java
package com.android.camera.activity;

import android.Manifest;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.android.camera.beauty.activity.BeautyFilterActivity;
import com.android.camera.beauty.activity.BeautyImageActivity;
import com.android.camera2.R;


public class MainActivity extends AppCompatActivity {

    private final String[] PERMISSIONS = new String[]{Manifest.permission.CAMERA};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_cam);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            requestPermissions(PERMISSIONS, 54321);
        }
        initView();
    }

    private void initView() {
        Button btnFace = findViewById(R.id.btn_face);
        btnFace.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, FaceBeautyActivity.class)));

        Button btnFilter = findViewById(R.id.btn_filter);
        btnFilter.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, BeautyFilterActivity.class)));

        Button btnPhoto = findViewById(R.id.btn_photo);
        btnPhoto.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, BeautyImageActivity.class)));
    }

}

File: app/src/main/java/com/android/camera2/MainActivity.kt
package com.android.camera2

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.android.camera.activity.MainActivity
import com.android.camera2.fragment.Camera2M1

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

//        addFragment()
        gotoCameraBeauty()
    }

    fun addFragment(){
        val frag = Camera2M1()

        supportFragmentManager.beginTransaction()
            .add(R.id.main, frag)
            .commit()
    }

    fun gotoCameraBeauty(){
        startActivity(Intent(this, MainActivity::class.java))
    }

}